File: backend/app/schemas/proposal.py
"""方案相关 Schema"""
from datetime import datetime, date
from pydantic import BaseModel, Field


class ProposalCreateRequest(BaseModel):
    """发起方案请求 —— 方案无校验环节，仅需设计人 + 审批人"""
    name: str = Field(..., min_length=1, max_length=100, description="方案名称")
    description: str | None = Field(None, max_length=500, description="补充说明")
    organization_id: int = Field(..., description="所属组织")
    designer_id: int = Field(..., description="设计人（工作节点负责人）")
    approvers: list[dict] = Field(..., description="审批人列表 [{'user_id': N}]")
    deadline: datetime | None = Field(None, description="截止日期")


class ProposalListItem(BaseModel):
    """方案列表项"""
    id: int
    name: str
    description: str | None = None
    organization_id: int
    initiator_id: int
    initiator_name: str = ""
    status: str
    created_at: datetime | None = None

    model_config = {"from_attributes": True}

File: frontend/src/api/proposal.ts
/** 方案 API */
import request from './request'
import type { PaginatedResponse } from './index'

export interface ProposalListItem {
  id: number
  name: string
  description: string | null
  organization_id: number
  organization_name: string  // 所属组织名称
  initiator_id: number
  initiator_name: string
  status: string
  created_at: string | null
}

export interface ProposalOrgCardItem {
  id: number
  name: string
  total_count: number
  running_count: number
  completed_count: number
  terminated_count: number
  latest_update_time: string | null
  is_current_user_org: boolean  // 是否为当前用户所属组织
}

export interface ProposalCreateRequest {
  name: string
  description?: string | null
  organization_id: number
  /** 设计人（工作节点负责人） */
  designer_id: number
  /** 方案无校验环节，仅需审批人 */
  approvers: { user_id: number }[]
  deadline?: string | null
}

/** 获取方案列表 */
export async function getProposals(params: {
  organization_id?: number
  status?: string
  keyword?: string
  page?: number
  page_size?: number
}) {
  const res = await request.get('/proposals', { params })
  return res.data as PaginatedResponse<ProposalListItem>
}

/** 获取方案组织卡片数据 */
export async function getProposalOrganizations(): Promise<{ organizations: ProposalOrgCardItem[] }> {
  const res = await request.get('/proposals/organizations')
  return res.data
}

/** 发起方案 */
export async function createProposal(data: ProposalCreateRequest) {
  const res = await request.post('/proposals', data)
  return res.data as { id: number; name: string; status: string }
}
